### PR TITLE
Fix DataFrame::select_columns to handle column names with a period

### LINKED ISSUE #3733: `DataFrame::select_columns` doesn't work with names containing "."
**Describe the bug**

**To Reproduce**
Put in `core/src/dataframe.rs`

```rust
    #[tokio::test]
    async fn select_with_periods() -> Result<()> {
        // define data with a column name that has a "." in it:
        let array: Int32Array = [1, 10].into_iter().collect();
        let batch =
            RecordBatch::try_from_iter(vec![("f.c1", Arc::new(array) as _)]).unwrap();

        let ctx = SessionContext::new();
        ctx.register_batch("t", batch).unwrap();

        let df = ctx
            .table("t")
            .unwrap()
            .select_columns(&["f.c1"])
            .unwrap();

        let df_results = df.collect().await.unwrap();

        assert_batches_sorted_eq!(
            vec![
                "+------+",
                "| f.c1 |",
                "+------+",
                "| 1    |",
                "| 10   |",
                "+------+",
            ],
            &df_results
        );

        Ok(())
}
```

Fails like:
```
thread 'dataframe::tests::select_with_periods' panicked at 'called `Result::unwrap()` on an `Err` value: SchemaError(FieldNotFound { qualifier: Some("f"), name: "c1", valid_fields: Some(["t.f.c1"]) })', datafusion/core/src/dataframe.rs:1437:14
```

**Expected behavior**
The test should pass

**Additional context**
Found while working on https://github.com/apache/arrow-datafusion/pull/3700

cc @hengfeiyang

### PR DESCRIPTION
**Pull request by zhoudongyan**
# Which issue does this PR close?

Closes #3733 

 # Rationale for this change

# What changes are included in this PR?

Changes: do not use col(), which tries to parse strings as identifiers (e.g. splits on .)

# Are there any user-facing changes?

One fewer bug

### CODE REVIEW

File: datafusion/core/src/dataframe.rs [hunk 1]
```diff
@@ -125,7 +125,10 @@ impl DataFrame {
             .iter()
             .map(|name| self.plan.schema().field_with_unqualified_name(name))
             .collect::<Result<Vec<_>>>()?;
-        let expr: Vec<Expr> = fields.iter().map(|f| col(f.name())).collect();
+        let expr: Vec<Expr> = fields
```
Comment (alamb) [inline datafusion/core/src/dataframe.rs:128]: 👍

**Review by alamb — APPROVED** [review]
Looks great -- thanks @zhoudongyan

**Comment by zhoudongyan** [comment]
@alamb May I know how can I fix the failed check so I can merge this PR?

**Comment by alamb** [comment]
Thanks @zhoudongyan  -- that was a great first contribution

**Comment by zhoudongyan** [comment]
@alamb Thanks for your review

**Comment by alamb** [comment]
@zhoudongyan thank you for the contribution!

**Pull request merged**